File: libs/hello-js/src/hello.ts
/**
 * Returns a hello string.
 *
 * @param name - The name to include. Optional.
 * @returns hello string.
 */
export function hello(name?: string | undefined): string {
  // eslint-disable-next-line @typescript-eslint/prefer-nullish-coalescing -- Handle empty string
  const nameToReturn = name || `world`;
  return `Hello, ${nameToReturn}!`;
}

File: libs/hello-cs/main/Hello.cs
namespace Connorjs.NxPolyglotExample.Hello;

public class Hello
{
    /// <summary>
    /// Returns a hello string.
    /// </summary>
    /// <param name="name">The name to include. Optional.</param>
    /// <returns>Hello string.</returns>
    public static string hello(string? name)
    {
        var nameToReturn = string.IsNullOrEmpty(name) ? "world" : name;
        return $"Hello, {nameToReturn}!";
    }
}
